salt/states/file.py

```diff
@@ -5663,7 +5663,7 @@ def copy(
     if not os.path.isdir(dname):
         if makedirs:
             try:
-                _makedirs(name=name)
+                _makedirs(name=name, user=user, group=group)
```

Comment:
what about 'mode'? should that be added too?

Reply:
Additional question: salt/states/file.py also contains a lot more of those `_makedirs(name=name)` calls. Shouldn't those also be called with almost all parameters?

(provided that _makedirs does the right thing)

Reply:
@srkunze i looked at the other `_makedirs` calls and those functions do not include the `user/group/mode` arguments to the function so I think for this PR just adding it for `file.copy` will be fine and if we want to add the user/group arguments to the other functions that can be handled in another PR in develop.

also nice catch @rares-pop i'll work on adding that